fdbrpc/sim2.actor.cpp

```diff
@@ -223,6 +228,27 @@ struct SimClogging {
 		auto& u = clogPairUntil[std::make_pair(from, to)];
 		u = std::max(u, now() + t);
 	}
+
+	void unclogPair(const IPAddress& from, const IPAddress& to) {
+		auto pair = std::make_pair(from, to);
+		auto it = clogPairUntil.find(pair);
+		if (it != clogPairUntil.end()) {
+			clogPairUntil.erase(it);
+		}
+
+		auto i = clogPairLatency.find(std::make_pair(from, to));
+		if (i != clogPairLatency.end()) {
+			clogPairLatency.erase(i);
+		}
+	}
+
+	// Clog a pair of processes until a time. This is more fine-grained than
+	// the IPAddress based one.
+	void clogPairFor(const NetworkAddress& from, const NetworkAddress& to, double t) {
+		auto& u = clogProcessPairUntil[std::make_pair(from, to)];
+		u = std::max(u, now() + t);
```

Comment:
I know this tries to be consistent with other functions, but it still looks weird. Can we just do

`clogProcessPairUntil[std::make_pair(from, to)] = std::max(u, now() + t);` ?

Easier to read in this way.

Reply:
Not exactly. Your proposal should be:
`clogProcessPairUntil[std::make_pair(from, to)] = std::max(clogProcessPairUntil[std::make_pair(from, to)], now() + t);`
which seems cumbersome to read.

Reply:
got it. Make sense.